### PR TITLE
[tt.dot_scaled]: Support for scale on operand "B"

### PR DESCRIPTION
**Pull request by etiotto**
This PR extends the support for `tt.dot_scaled` decomposition pass and allows the "B" operand to have a scaling factor (as well as the "A" operand).

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUTransforms/AccelerateMatmul.cpp [hunk 1]
```diff
@@ -236,69 +232,118 @@ class DecomposeScaledBlocked : public OpRewritePattern<tt::DotScaledOp> {
...
+
+    assert((bDesc.scale && !aDesc.scale) && "NYI: both LHS and RHS scale");
+    TensorValue newB =
+        convertScaledOperand<1>(bDesc, dpasEnc, newRetType, mod, rewriter);
+    TensorValue newA =
+        convertUnscaledOperand<0>(aDesc, dpasEnc, newRetType, rewriter);
+    return {newA, newB};
+  }
+
+  template <int opIdx>
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/AccelerateMatmul.cpp:281]:
```suggestion
  template <unsigned opIdx>
```

**Review by whitneywhtsang — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUTransforms/AccelerateMatmul.cpp [hunk 2]
```diff
@@ -236,69 +232,118 @@ class DecomposeScaledBlocked : public OpRewritePattern<tt::DotScaledOp> {
...
+    if (aDesc.scale) {
+      assert(bDesc.scale == nullptr && "NYI: both LHS and RHS scale");
+      TensorValue newA =
+          convertScaledOperand<0>(aDesc, dpasEnc, newRetType, mod, rewriter);
+      TensorValue newB =
+          convertUnscaledOperand<1>(bDesc, dpasEnc, newRetType, rewriter);
+      return {newA, newB};
+    }
+
+    assert((bDesc.scale && !aDesc.scale) && "NYI: both LHS and RHS scale");
```
Comment (LiyangLingIntel) [inline third_party/intel/lib/TritonIntelGPUTransforms/AccelerateMatmul.cpp:273]: From the test cases, operand A would be either FP4 or FP8, will there be a case that we only do scaling on operand B?

**Review by LiyangLingIntel — APPROVED** [review]
LGTM. 
FYI. I'm tryting to use Blocked Layout for UpcastMXFPOp to accelerate the enabling process as the first step. It would be easier to understand and debug, but need to change some of the AccelerateMatmul conversion code.

**Comment by etiotto** [comment]
> LGTM. FYI. I'm tryting to use Blocked Layout for UpcastMXFPOp to accelerate the enabling process as the first step. It would be easier to understand and debug, but need to change some of the AccelerateMatmul conversion code.

That's OK.  You can change the AccelerateMatmul code with a subsequent PR (after this one is merged) but please keep this code (perhaps under an option) so that we can continue to the lit tests I have added. Eventually we will want the result of the upcast_mxfp to have dot layout (with DPAS parent layout) for best performance.

**Pull request merged**